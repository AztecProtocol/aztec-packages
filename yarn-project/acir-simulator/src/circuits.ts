import { AztecAddress, Fr, FunctionData, PrivateCircuitPublicInputs, TxContext } from '@aztec/circuits.js';

export class TxRequest {
  constructor(
    public readonly from: AztecAddress,
    public readonly to: AztecAddress,
    public readonly functionData: FunctionData,
    public readonly args: Fr[],
    public readonly txContext: TxContext,
    public readonly nonce: Fr,
    public readonly chainId: Fr,
  ) {}

  toBuffer() {
    return Buffer.alloc(0);
  }
}

export class PrivateCallStackItem {
  constructor(
    public readonly contractAddress: AztecAddress,
    public readonly functionSelector: number,
    public readonly publicInputs: PrivateCircuitPublicInputs,
  ) {}
}
