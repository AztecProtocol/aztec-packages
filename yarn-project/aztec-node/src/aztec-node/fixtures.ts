import {
  AccumulatedData,
  AffineElement,
  AggregationObject,
  ConstantData,
  ContractDeploymentData,
  EMITTED_EVENTS_LENGTH,
  EthAddress as CircuitEthAddress,
  Fq,
  Fr,
  FunctionData,
  KERNEL_L1_MSG_STACK_LENGTH,
  KERNEL_NEW_COMMITMENTS_LENGTH,
  KERNEL_NEW_CONTRACTS_LENGTH,
  KERNEL_NEW_NULLIFIERS_LENGTH,
  KERNEL_OPTIONALLY_REVEALED_DATA_LENGTH,
  KERNEL_PRIVATE_CALL_STACK_LENGTH,
  KERNEL_PUBLIC_CALL_STACK_LENGTH,
  NewContractData,
  OldTreeRoots,
  OptionallyRevealedData,
  PrivateKernelPublicInputs,
  TxContext,
  UInt8Vector,
} from '@aztec/circuits.js';
import { EthereumRpc } from '@aztec/ethereum.js/eth_rpc';
import { WalletProvider } from '@aztec/ethereum.js/provider';
import { AztecAddress, randomBytes } from '@aztec/foundation';
import { Rollup, Yeeter } from '@aztec/l1-contracts';
import { Tx } from '@aztec/tx';

// REFACTOR: Move deployment logic to l1-contracts package, and refactor it out of other integration tests (archiver, sequencer)
export const deployRollupContract = async (provider: WalletProvider, ethRpc: EthereumRpc) => {
  const deployAccount = provider.getAccount(0);
  const contract = new Rollup(ethRpc, undefined, { from: deployAccount, gas: 1000000 });
  await contract.deploy().send().getReceipt();
  return contract.address;
};

export const deployYeeterContract = async (provider: WalletProvider, ethRpc: EthereumRpc) => {
  const deployAccount = provider.getAccount(0);
  const contract = new Yeeter(ethRpc, undefined, { from: deployAccount, gas: 1000000 });
  await contract.deploy().send().getReceipt();
  return contract.address;
};

export const createProvider = (host: string, mnemonic: string, accounts: number) => {
  const walletProvider = WalletProvider.fromHost(host);
  walletProvider.addAccountsFromMnemonic(mnemonic, accounts);
  return walletProvider;
};

// REFACTOR: Use @aztec/circuit.js/factories where possible
export const createCircuitEthAddress = () => {
  return new CircuitEthAddress(randomBytes(20));
};

export const createRandomCommitments = (num: number) => {
  return Array(num)
    .fill(0)
    .map(() => new Fr(randomBytes(32)));
};

export const createOptionallyRetrievedData = () => {
  const func = new FunctionData(0, true, true);
  return new OptionallyRevealedData(
    new Fr(0),
    func,
    createRandomCommitments(EMITTED_EVENTS_LENGTH),
    new Fr(0),
    createCircuitEthAddress(),
    true,
    true,
    true,
    true,
  );
};

export const createOptionallyRetrievedDatas = (num: number) => {
  return Array(num)
    .fill(0)
    .map(() => createOptionallyRetrievedData());
};

export const createNewContractData = () => {
  return new NewContractData(AztecAddress.random(), createCircuitEthAddress(), new Fr(randomBytes(32)));
};

export const createNewContractDatas = (num: number) => {
  return Array(num)
    .fill(0)
    .map(() => createNewContractData());
};

export const createTx = () => {
  const oldTreeRoots = new OldTreeRoots(new Fr(0), new Fr(0), new Fr(0), new Fr(0));
  const contractDeploymentData = new ContractDeploymentData(
    new Fr(randomBytes(32)),
    new Fr(randomBytes(32)),
    new Fr(randomBytes(32)),
    createCircuitEthAddress(),
  );
  const txContext = new TxContext(false, false, true, contractDeploymentData);
  const constantData = new ConstantData(oldTreeRoots, txContext);
  const aggregationObject = new AggregationObject(
    new AffineElement(new Fq(0), new Fq(0)),
    new AffineElement(new Fq(0), new Fq(0)),
    [],
    [],
    false,
  );
  const accumulatedData = new AccumulatedData(
    aggregationObject,
    new Fr(0),
    createRandomCommitments(KERNEL_NEW_COMMITMENTS_LENGTH),
    createRandomCommitments(KERNEL_NEW_NULLIFIERS_LENGTH),
    createRandomCommitments(KERNEL_PRIVATE_CALL_STACK_LENGTH),
    createRandomCommitments(KERNEL_PUBLIC_CALL_STACK_LENGTH),
    createRandomCommitments(KERNEL_L1_MSG_STACK_LENGTH),
    createNewContractDatas(KERNEL_NEW_CONTRACTS_LENGTH),
    createOptionallyRetrievedDatas(KERNEL_OPTIONALLY_REVEALED_DATA_LENGTH),
  );
  const kernelInputs = new PrivateKernelPublicInputs(accumulatedData, constantData, true);
  return new Tx(kernelInputs, new UInt8Vector(Buffer.alloc(0)));
};
