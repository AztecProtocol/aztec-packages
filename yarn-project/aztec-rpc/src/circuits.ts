import { randomBytes } from '@aztec/foundation';

export class Signature {
  public static SIZE = 64;

  public static random() {
    return new Signature(randomBytes(Signature.SIZE));
  }

  constructor(public readonly buffer: Buffer) {}
}
