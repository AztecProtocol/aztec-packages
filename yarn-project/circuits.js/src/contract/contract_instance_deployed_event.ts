import { AztecAddress } from '@aztec/foundation/aztec-address';
import { toBigIntBE } from '@aztec/foundation/bigint-buffer';
import { EthAddress } from '@aztec/foundation/eth-address';
import { Fr } from '@aztec/foundation/fields';
import { BufferReader } from '@aztec/foundation/serialize';
import { ContractInstanceWithAddress } from '@aztec/types/contracts';

import { DEPLOYER_CONTRACT_INSTANCE_DEPLOYED_MAGIC_VALUE } from '../constants.gen.js';

/** Event emitted from the ContractInstanceDeployer. */
export class ContractInstanceDeployedEvent {
  constructor(
    public readonly address: AztecAddress,
    public readonly version: number,
    public readonly salt: Fr,
    public readonly contractClassId: Fr,
    public readonly initializationHash: Fr,
    public readonly portalContractAddress: EthAddress,
    public readonly publicKeysHash: Fr,
    public readonly universalDeploy: boolean,
  ) {}

  static isContractInstanceDeployedEvent(log: Buffer) {
    return toBigIntBE(log.subarray(0, 32)) == DEPLOYER_CONTRACT_INSTANCE_DEPLOYED_MAGIC_VALUE;
  }

  // TODO(@spalladino) We should be loading the singleton address from protocol-contracts,
  // but the protocol-contracts package depends on this one, and we cannot have circular dependencies,
  // hence we require it as an argument for now.
  static fromLogs(logs: { contractAddress: AztecAddress; data: Buffer }[], instanceDeployerAddress: AztecAddress) {
    return logs
      .filter(log => ContractInstanceDeployedEvent.isContractInstanceDeployedEvent(log.data))
      .filter(log => log.contractAddress.equals(instanceDeployerAddress))
      .map(log => ContractInstanceDeployedEvent.fromLogData(log.data));
  }

  static fromLogData(log: Buffer) {
    if (!this.isContractInstanceDeployedEvent(log)) {
      const magicValue = DEPLOYER_CONTRACT_INSTANCE_DEPLOYED_MAGIC_VALUE.toString(16);
      throw new Error(`Log data for ContractInstanceDeployedEvent is not prefixed with magic value 0x${magicValue}`);
    }
    const reader = new BufferReader(log.subarray(32));
    const address = reader.readObject(AztecAddress);
    const version = reader.readObject(Fr).toNumber();
    const salt = reader.readObject(Fr);
    const contractClassId = reader.readObject(Fr);
    const initializationHash = reader.readObject(Fr);
    const portalContractAddress = EthAddress.fromField(reader.readObject(Fr));
    const publicKeysHash = reader.readObject(Fr);
    const universalDeploy = reader.readObject(Fr).toBool();

    return new ContractInstanceDeployedEvent(
      address,
      version,
      salt,
      contractClassId,
      initializationHash,
      portalContractAddress,
      publicKeysHash,
      universalDeploy,
    );
  }

  toContractInstance(): ContractInstanceWithAddress {
    if (this.version !== 1) {
      throw new Error(`Unexpected contract instance version ${this.version}`);
    }

    return {
      address: this.address,
      version: this.version,
      contractClassId: this.contractClassId,
      initializationHash: this.initializationHash,
      portalContractAddress: this.portalContractAddress,
      publicKeysHash: this.publicKeysHash,
      salt: this.salt,
    };
  }
}
