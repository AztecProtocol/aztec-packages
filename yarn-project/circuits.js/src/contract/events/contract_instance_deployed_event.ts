import { AztecAddress } from '@aztec/foundation/aztec-address';
import { toBigIntBE } from '@aztec/foundation/bigint-buffer';
import { EthAddress } from '@aztec/foundation/eth-address';
import { Fr } from '@aztec/foundation/fields';
import { BufferReader } from '@aztec/foundation/serialize';
import { type ContractInstanceWithAddress } from '@aztec/types/contracts';

import { DEPLOYER_CONTRACT_ADDRESS, DEPLOYER_CONTRACT_INSTANCE_DEPLOYED_MAGIC_VALUE } from '../../constants.gen.js';

/** Event emitted from the ContractInstanceDeployer. */
export class ContractInstanceDeployedEvent {
  constructor(
    public readonly address: AztecAddress,
    public readonly version: number,
    public readonly salt: Fr,
    public readonly contractClassId: Fr,
    public readonly initializationHash: Fr,
    public readonly portalContractAddress: EthAddress,
    public readonly publicKeysHash: Fr,
    public readonly deployer: AztecAddress,
  ) {}

  static isContractInstanceDeployedEvent(log: Buffer) {
    return toBigIntBE(log.subarray(0, 32)) == DEPLOYER_CONTRACT_INSTANCE_DEPLOYED_MAGIC_VALUE;
  }

  static fromLogs(logs: { contractAddress: AztecAddress; data: Buffer }[]) {
    return logs
      .filter(log => ContractInstanceDeployedEvent.isContractInstanceDeployedEvent(log.data))
      .filter(log => log.contractAddress.equals(AztecAddress.fromBigInt(DEPLOYER_CONTRACT_ADDRESS)))
      .map(log => ContractInstanceDeployedEvent.fromLogData(log.data));
  }

  static fromLogData(log: Buffer) {
    if (!this.isContractInstanceDeployedEvent(log)) {
      const magicValue = DEPLOYER_CONTRACT_INSTANCE_DEPLOYED_MAGIC_VALUE.toString(16);
      throw new Error(`Log data for ContractInstanceDeployedEvent is not prefixed with magic value 0x${magicValue}`);
    }
    const reader = new BufferReader(log.subarray(32));
    const address = reader.readObject(AztecAddress);
    const version = reader.readObject(Fr).toNumber();
    const salt = reader.readObject(Fr);
    const contractClassId = reader.readObject(Fr);
    const initializationHash = reader.readObject(Fr);
    const portalContractAddress = EthAddress.fromField(reader.readObject(Fr));
    const publicKeysHash = reader.readObject(Fr);
    const deployer = reader.readObject(AztecAddress);

    return new ContractInstanceDeployedEvent(
      address,
      version,
      salt,
      contractClassId,
      initializationHash,
      portalContractAddress,
      publicKeysHash,
      deployer,
    );
  }

  toContractInstance(): ContractInstanceWithAddress {
    if (this.version !== 1) {
      throw new Error(`Unexpected contract instance version ${this.version}`);
    }

    return {
      address: this.address,
      version: this.version,
      contractClassId: this.contractClassId,
      initializationHash: this.initializationHash,
      portalContractAddress: this.portalContractAddress,
      publicKeysHash: this.publicKeysHash,
      salt: this.salt,
      deployer: this.deployer,
    };
  }
}
