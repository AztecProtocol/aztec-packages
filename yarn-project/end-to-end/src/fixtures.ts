export const MNEMONIC = 'test test test test test test test test test test test junk';
