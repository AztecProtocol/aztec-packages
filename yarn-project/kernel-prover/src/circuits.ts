// See aztec3/constants.hpp
// Copied here for prototyping purposes

import { AztecAddress, Fr, FunctionData, TxContext } from '@aztec/circuits.js';

export class TxRequest {
  constructor(
    public readonly from: AztecAddress,
    public readonly to: AztecAddress,
    public readonly functionData: FunctionData,
    public readonly args: Fr[],
    public readonly txContext: TxContext,
    public readonly nonce: Fr,
    public readonly chainId: Fr,
  ) {}

  toBuffer() {
    return Buffer.alloc(0);
  }
}

export class Signature {
  public static SIZE = 64;

  constructor(public readonly buffer: Buffer) {}
}
