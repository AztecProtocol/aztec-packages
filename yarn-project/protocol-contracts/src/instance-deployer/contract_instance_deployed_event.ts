import { type ContractInstanceWithAddress, type PrivateLog, PublicKeys } from '@aztec/circuits.js';
import { AztecAddress } from '@aztec/foundation/aztec-address';
import { Fr } from '@aztec/foundation/fields';
import { BufferReader } from '@aztec/foundation/serialize';

import { DEPLOYER_CONTRACT_INSTANCE_DEPLOYED_TAG } from '../protocol_contract_data.js';

/** Event emitted from the ContractInstanceDeployer. */
export class ContractInstanceDeployedEvent {
  constructor(
    public readonly address: AztecAddress,
    public readonly version: number,
    public readonly salt: Fr,
    public readonly contractClassId: Fr,
    public readonly initializationHash: Fr,
    public readonly publicKeys: PublicKeys,
    public readonly deployer: AztecAddress,
  ) {}

  static isContractInstanceDeployedEvent(log: PrivateLog) {
    return log.fields[0].equals(DEPLOYER_CONTRACT_INSTANCE_DEPLOYED_TAG);
  }

  static fromLog(log: PrivateLog) {
    const bufferWithoutTag = log.toBuffer().subarray(32);
    const reader = new BufferReader(bufferWithoutTag);
    const address = reader.readObject(AztecAddress);
    const version = reader.readObject(Fr).toNumber();
    const salt = reader.readObject(Fr);
    const contractClassId = reader.readObject(Fr);
    const initializationHash = reader.readObject(Fr);
    const publicKeys = reader.readObject(PublicKeys);
    const deployer = reader.readObject(AztecAddress);

    return new ContractInstanceDeployedEvent(
      address,
      version,
      salt,
      contractClassId,
      initializationHash,
      publicKeys,
      deployer,
    );
  }

  toContractInstance(): ContractInstanceWithAddress {
    if (this.version !== 1) {
      throw new Error(`Unexpected contract instance version ${this.version}`);
    }

    return {
      address: this.address,
      version: this.version,
      contractClassId: this.contractClassId,
      initializationHash: this.initializationHash,
      publicKeys: this.publicKeys,
      salt: this.salt,
      deployer: this.deployer,
    };
  }
}
