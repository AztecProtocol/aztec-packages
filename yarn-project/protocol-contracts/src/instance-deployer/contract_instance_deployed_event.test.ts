import { PrivateLog } from '@aztec/circuits.js';

import { getSampleContractInstanceDeployedEventPayload } from '../tests/fixtures.js';
import { ContractInstanceDeployedEvent } from './contract_instance_deployed_event.js';

describe('ContractInstanceDeployedEvent', () => {
  it('parses an event as emitted by the ClassInstanceDeployer', () => {
    const data = getSampleContractInstanceDeployedEventPayload();
    const log = PrivateLog.fromBuffer(data);
    expect(ContractInstanceDeployedEvent.isContractInstanceDeployedEvent(log)).toBe(true);

    const event = ContractInstanceDeployedEvent.fromLog(log);
    expect(event.address.toString()).toEqual('0x0c5c6978e380c4e3940ab74770639260bcc75c93c3d0ae48ee4a241d555b094e');
    expect(event.contractClassId.toString()).toEqual(
      '0x2b78af6d543573f77372e53e66932714d68877b4bcbb18671e68a846795297e1',
    );
  });
});
