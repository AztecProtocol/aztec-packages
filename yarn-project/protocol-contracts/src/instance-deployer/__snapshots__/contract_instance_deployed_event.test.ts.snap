// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContractInstanceDeployedEvent parses an event as emitted by the ClassInstanceDeployer 1`] = `
ContractInstanceDeployedEvent {
  "address": "0x2b340e606093cce3fb3378ab30f0068d5b9e25a44030b2a831a6a7445ab028f7",
  "contractClassId": "0x174515a1d42a4dd065fc4031e2b240a962987197e8bf55d8c7b927978ff9ce38",
  "deployer": "0x0000000000000000000000000000000000000000000000000000000000000000",
  "initializationHash": "0x129873d1e20601e6fc4a2df1f458bc0541e0a04bcfd35ebdae5f98893c36b930",
  "publicKeys": PublicKeys {
    "masterIncomingViewingPublicKey": "0x022104824c313ae3c071d752dba0ceab9f5c8c4b73ab3ebe9b1624a8405e16ca0f76402b0f09bc9097bfcb2589bea40089c54412d7bc5c464557fabb51aed6ed",
    "masterNullifierPublicKey": "0x02418fd9e36b1e0df67ce0a53be35cd6dd87fd0a15595221fc82620c54d5996617c3d818bfb95e09b8cda8fad2044c5f27196ad4564a2a711da4691d98b70037",
    "masterOutgoingViewingPublicKey": "0x02bbad84b1c58ac925f887d1ae08cc4cc273d5130b29eb6662aa162d5a28884c0c746eba643fbe25637632922f527d1d1236ff790378c97213cca35839dad76a",
    "masterTaggingPublicKey": "0x16617bba076cdd3426e10263481a998c4c275b283aaf1f42539900e3a9efa2bf24b5e4341a949daa0bde1611263777e4aa468f6da73d2f884a499135428c201a",
  },
  "salt": "0x1c734816f7263dba6df121e001323d7b461a17b144b383f5d4add1ec7a65b5d5",
  "version": 1,
}
`;
